### PR TITLE
Battle Maps docs

### PR DESCRIPTION
**Pull request by jmks**
Fix for #513 

* Fixed some of the examples/tests. Also provided a few `false` examples
* Updated the Example solution to match the solution (the actual solution seems fine)
* Run the examples as a moduledoc test

The last commit is a bit speculative. I saw it in a blog post once and always wanted to try it :smile: 
It seems to only see changes when you recompile before running the tests.
So a developer workflow would be like `mix clean && mix test` which is a bit awkward (but could be automated).
However, it would work great in CI.

What do you think?

### CODE REVIEW

File: utils/lib/solutions.ex [hunk 1]
```diff
@@ -1026,11 +1026,17 @@ defmodule Utils.Solutions do
     defstruct []
   end
 
-  defprotocol BattleCharacter do
+  defprotocol Character do
+    @moduledoc File.read!(Path.expand("../exercises/battle_map.livemd"))
```
Comment (BrooklinJazz) [inline utils/lib/solutions.ex:1030]: Livebook is working on a feature to automatically run any doctests in a module, so we're leaning towards documenting modules in the exercise itself in anticipation of that feature arriving in Livebook. 

We're moving away from using the solutions module and the `utils` project entirely except for running scripts and tests. This way livebooks can be run without depending on the `utils` project.

Comment (BrooklinJazz) [reply utils/lib/solutions.ex:1030]: Very cool idea though reading the docs from the livemd and ensuring they remain in sync with the solution.

Comment (BrooklinJazz) [reply utils/lib/solutions.ex:1030]: Can you remove the character protocol from Solutions? Then this looks good to merge, well done!

**Review by BrooklinJazz — CHANGES_REQUESTED** [review]
Well done! We're moving away from using the solutions module, so we can remove the BattleMap related code from there then we're good to merge.

**Comment by BrooklinJazz** [comment]
Looks like a test failed. Make sure to run `mix format`. There's also a `mix bc` script that runs a bunch of scripts related to the project. I don't think that's what's causing the issue but if the test still fails after formatting that's what I would try next.

**Comment by BrooklinJazz** [comment]
Hmm, for some reason, there's still a `mix format` issue. If running `mix format` again doesn't work let me know and I can help debug. 🤔

File: utils/lib/solutions.ex (hunk 1 above)
Comment (jmks) [reply utils/lib/solutions.ex:1030]: Having the tests run in Livebook would be great!

**Comment by BrooklinJazz** [comment]
Thank you!

**Pull request merged**